experimental/gcp-log/internal/storage/storage.go

```diff
@@ -376,14 +377,45 @@ func (c *Client) Sequence(ctx context.Context, leafhash []byte, leaf []byte) (ui
 			return 0, fmt.Errorf("couldn't create leafhash object: %w", err)
 		}
 		if err := wLeaf.Close(); err != nil {
-			return 0, fmt.Errorf("couldn't close writer for object %q", leafPath)
+			return 0, fmt.Errorf("couldn't close writer for object %q, %w", leafPath, err)
 		}
 
 		// All done!
 		return seq, nil
 	}
 }
 
+// assertContent checks that the content at `gcsPath` matches the passed in `data`.
+func (c *Client) assertContent(ctx context.Context, gcsPath string, data []byte) (equal bool, err error) {
+	bkt := c.gcsClient.Bucket(c.bucket)
+
+	obj := bkt.Object(gcsPath)
+	r, err := obj.NewReader(ctx)
+	if err != nil {
+		if errors.Is(err, gcs.ErrObjectNotExist) {
+			// Return the generic NotExist error so that tileCache.Visit can differentiate
+			// between this and other errors.
+			return false, err
```

Comment:
This seems to return the same `err` as below (rather than a `NotExist`) is that intended, or could the `if` block just be removed?

Reply:
Thanks, this whole block was unnecessary. (Careless copy and paste.)